skbeam/core/accumulators/binned_statistic.py

```diff
@@ -130,7 +136,13 @@ def __init__(self, sample, statistic='mean',
         # Compute the bin number each sample falls into.
         Ncount = {}
         for i in np.arange(self.D):
-            Ncount[i] = np.digitize(sample[:, i], self.edges[i])
+            # Apply mask in a non-ideal way by setting value outside range.
+            # Would be better to do this using bincount "weights", perhaps.
+            thissample = sample[:, i]
```

Comment:
This code is adapted from source in scipy, I would trust them.

I think transposing _may_ induce a copy if numpy can not sort out a way to compute the required strides for the transposed view.